### PR TITLE
[638] Use automatic layout configuration in incremental layout

### PR DESCRIPTION
**Pull request by AxelRICHARD**
### Type of this PR 

- [x] Bug fix
- [ ] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

Bug: https://github.com/eclipse-sirius/sirius-components/issues/638

### What does this PR do?

Use Node and Edge Label Positions and Paddings from automatic layout in incremental layout.

### How to test this PR?

- [ ] Frontend Unit tests
- [x] Backend Unit tests
- [ ] Cypress : please specify
- [ ] Manual Test : please specify

### Checklist

- [x] I have read CONTRIBUTING carefully.
- [x] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [x] I have covered my changes by unit tests or integration tests or manual tests.
- [x] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java [hunk 1]
```diff
@@ -62,8 +63,10 @@ public IncrementalLayoutEngine(NodeSizeProvider nodeSizeProvider) {
         this.nodeSizeProvider = Objects.requireNonNull(nodeSizeProvider);
     }
 
-    public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramLayoutData diagram) {
+    public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramLayoutData diagram, LayoutConfigurator layoutConfigurator) {
         this.nodePositionProvider.reset();
+        this.nodeLabelPositionProvider.setLayoutConfigurator(layoutConfigurator);
+        this.edgeLabelPositionProvider.setLayoutConfigurator(layoutConfigurator);
```
Comment (sbegaudeau) [inline backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java:72]: Do not use a setter, give those arguments to the constructor instead

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java [hunk 2]
```diff
@@ -25,7 +29,20 @@
  */
 public class EdgeLabelPositionProvider {
 
+    private LayoutConfigurator layoutConfigurator;
```
Comment (sbegaudeau) [inline backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java:32]: Should be final

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java [hunk 3]
```diff
@@ -25,7 +29,20 @@
...
+
+    public void setLayoutConfigurator(LayoutConfigurator layoutConfigurator) {
+        this.layoutConfigurator = layoutConfigurator;
+    }
+
     public Position getCenterPosition(EdgeLayoutData edge, LabelLayoutData label) {
+        Double spacingEdgeLabel = null;
+        if (this.layoutConfigurator instanceof SiriusWebLayoutConfigurator) {
+            spacingEdgeLabel = ((SiriusWebLayoutConfigurator) this.layoutConfigurator).configure(ElkEdge.class).getProperty(CoreOptions.SPACING_EDGE_LABEL);
+        }
```
Comment (sbegaudeau) [inline backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java:42]: Why does this need to be casted to SiriusWebLayoutConfigurator?

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java (hunk 2 above)
Comment (sbegaudeau) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java:32]: Do we really need to use LayoutConfigurator if we are casting it so many time to SiriusWebLayoutConfigurator?

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/NodeLabelPositionProvider.java [hunk 4]
```diff
@@ -41,12 +170,22 @@ public Position getPosition(NodeLayoutData node, LabelLayoutData label) {
...
+        boolean outside = false;
+        for (NodeLabelPlacement nodeLabelPlacement : nodeLabelPlacementSet) {
+            switch (nodeLabelPlacement) {
+            case OUTSIDE:
+                outside = true;
+                break;
+            default:
+                break;
+            }
+        }
```
Comment (sbegaudeau) [inline backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/NodeLabelPositionProvider.java:188]: There are a lot of `for` / `switch` in this PR which does not make the code very easy to read. Here this code could easily be replaced by a `nodeLabelPlacementSet.stream().anyMatch(NodeLabelPlacement.OUTSIDE::equals)`.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java (hunk 1 above)
Comment (AxelRICHARD) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java]: Done

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java (hunk 2 above)
Comment (AxelRICHARD) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java]: Done

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/NodeLabelPositionProvider.java (hunk 4 above)
Comment (AxelRICHARD) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/NodeLabelPositionProvider.java]: Done

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java (hunk 3 above)
Comment (AxelRICHARD) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/EdgeLabelPositionProvider.java]: Done

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**